protocol/chat.go

```diff
@@ -391,6 +398,14 @@ func (c *Chat) UpdateFromMessage(message *common.Message, timesource common.Time
 	return nil
 }
 
+func (c *Chat) ShouldIncreaseMessageCount(message *common.Message) bool {
```

Comment:
I moved the checks in this function so that it can be reused in the AC service later if needed